### ISSUE TITLE
onnxruntime.capi.onnxruntime_pybind11_state.RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION 

### ISSUE THREAD

**Issue by neso613** [microsoft/onnxruntime#10657]
**Describe the bug**
Hi,
I am trying to write inference code for onnx model, stuck at this errror
`  File "/home/cogknit/anaconda3/envs/hinditts/lib/python3.7/site-packages/onnxruntime/capi/onnxruntime_inference_collection.py", line 192, in run
    return self._sess.run(output_names, input_feed, run_options)
onnxruntime.capi.onnxruntime_pybind11_state.RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Non-zero status code returned while running Mul node. Name:'Mul_2142' Status Message: /onnxruntime_src/onnxruntime/core/providers/cpu/math/element_wise_ops.h:497 void onnxruntime::BroadcastIterator::Init(ptrdiff_t, ptrdiff_t) axis == 1 || axis == largest was false. Attempting to broadcast an axis by a dimension other than 1. 194 by 202
`

This is models dimensions-
`Graph Input] name: input, shape: [0, 0]
[Graph Input] name: input_lengths, shape: [0]
[Graph Output] name: output, shape: [0, 0, 0]
`

`[name: "input"
type {
  tensor_type {
    elem_type: 7
    shape {
      dim {
        dim_param: "batch_size"
      }
      dim {
        dim_param: "phonemes"
      }
    }
  }
}
, name: "input_lengths"
type {
  tensor_type {
    elem_type: 7
    shape {
      dim {
        dim_param: "batch_size"
      }
    }
  }
}
]
[name: "output"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {
        dim_param: "batch_size"
      }
      dim {
        dim_param: "Muloutput_dim_1"
      }
      dim {
        dim_param: "Muloutput_dim_2"
      }
    }
  }
}
]
`

**Urgency**
Its a critical error. This is a bottelneck of important demo. If possible, try to resolve this ASAP

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 20.04
- ONNX Runtime installed from (source or binary): Binary
- ONNX Runtime version: 1.10.0
- Python version: 3.7

**Screenshots**
onnx [model](https://drive.google.com/file/d/1x09B_IaQoDltOYOcXvvD2jgiMkxoR_tW/view?usp=sharing)

**Comment by neso613** [comment]
sharing this model info using netron - 
![image](https://user-images.githubusercontent.com/31642462/155502598-c6b72129-eae5-4509-aa11-47fdcc01d09a.png)

**Comment by neso613** [comment]
@skomirishett

**Comment by yuslepukhin** [comment]
`The input tensor cannot be reshaped to the requested shape. Input shape:{1,25}, requested shape:{50}`

The total size of the tensor should match the new shape. In this case the input has size of 25, it can not become 50.

**Comment by neso613** [comment]
> `The input tensor cannot be reshaped to the requested shape. Input shape:{1,25}, requested shape:{50}`
> 
> The total size of the tensor should match the new shape. In this case the input has size of 25, it can not become 50.

But onnx model has dynamic axis. Why this cannot reshaped?
This is models dimensions-
Graph Input] name: input, shape: [0, 0] [Graph Input] name: input_lengths, shape: [0] [Graph Output] name: output, shape: [0, 0, 0]

**Comment by neso613** [comment]
looking forward to hear from community

**Comment by neso613** [comment]
@synesthesiam https://github.com/synesthesiam

**Comment by neso613** [comment]
This issue is basically - while exportin torch model to onnx, we have specified dummy_input and that dummy_input shape is somehow taken as fixed parameter to onnx model. Thats why it hold that dummy_shape as fixed and cannot taken dynamic input shape.

**Comment by neso613** [comment]
While analysing model graph, I found - 
![image](https://user-images.githubusercontent.com/31642462/157660744-0210c9f6-838d-4678-a3b6-781f489d34d0.png)

Here is the issue at this node -
![image](https://user-images.githubusercontent.com/31642462/157660797-602a0677-1d9c-4c02-b83d-b89307cf3fc4.png)

Please suggest some way for this error or this is a BUG.

**Comment by tqnwhz** [comment]
Hi bro, I also met this problem recently, searching for solutions. I'm wondering whether you found any solution or it's confirmed to be a bug. Thanks very much.

**Comment by neso613** [comment]
@tqnwhz No, I do not find any solution for this. I believe its a BUG.

**Comment by tqnwhz** [comment]
Hi, I read the torch onnx doc, finding that the default export mode (tracing) can't deal with control flows. I guess that it might be a cause?

**Comment by ynma-hanvo** [comment]
similar problem here:
-----------------------------------------------
onnxruntime.capi.onnxruntime_pybind11_state.RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Exception during initialization: /onnxruntime_src/onnxruntime/core/providers/cpu/tensor/reshape_helper.h:41 onnxruntime::ReshapeHelper::ReshapeHelper(const onnxruntime::TensorShape&, std::vector<long int>&, bool) gsl::narrow_cast<int64_t>(input_shape.Size()) == size was false
------------------------------------------------
my model exported with dummy input  ,   
the model is quite complex which takes in 2 images  and  4 outpus,
but when do inference with the same dummy input , got the above error.

**Comment by shifujain** [comment]
Facing a similar issue where an additional reshape node is getting added
![image](https://user-images.githubusercontent.com/38124154/172655468-d975b9ad-0a67-4c20-9ae0-bd2c6eb4ad30.png)

here is the problematic node
![image](https://user-images.githubusercontent.com/38124154/172655755-f4085285-688e-45f6-87c6-df91c0998c4b.png)

**Comment by rbavery** [comment]
I'm also facing this issue

**Comment by phamkhactu** [comment]
@rbavery Have you solved this issue?? I've encountered in it.

**Comment by renaissancelab** [comment]
I have met same problems with follow errors:

`2023-06-30 18:25:53.036 | ERROR    | roop.kafka_client:consume_message:89 - [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Exception during initialization: /onnxruntime_src/onnxruntime/core/providers/cuda/cuda_call.cc:121 std::conditional_t<THRW, void, onnxruntime::common::Status> onnxruntime::CudaCall(ERRTYPE, const char*, const char*, ERRTYPE, const char*, const char*, int) [with ERRTYPE = cublasStatus_t; bool THRW = true; std::conditional_t<THRW, void, onnxruntime::common::Status> = void] /onnxruntime_src/onnxruntime/core/providers/cuda/cuda_call.cc:114 std::conditional_t<THRW, void, onnxruntime::common::Status> onnxruntime::CudaCall(ERRTYPE, const char*, const char*, ERRTYPE, const char*, const char*, int) [with ERRTYPE = cublasStatus_t; bool THRW = true; std::conditional_t<THRW, void, onnxruntime::common::Status> = void] CUBLAS failure 3: CUBLAS_STATUS_ALLOC_FAILED ; GPU=0 ; hostname=tec-deepface-gpu-t4-centos7 ; file=/onnxruntime_src/onnxruntime/core/providers/cuda/cuda_execution_provider.cc ; line=164 ; expr=cublasCreate(&cublas_handle_);`

**Comment by isALEXme** [comment]
Yo, 

For my part I found a solution, do not touch dummy_input : 

def export_to_onnx(dummy_input):  # Assuming a default image size of 256x256
    # with 3 channels
    global model
    model.eval()
    # Solution : USED TO HAVE THIS LINE 
    # Export the model
    dummy_input = (dummy_input / 255.0).float()
    torch.onnx.export(
        model,  # model being run
        dummy_input,  # model input (or a tuple for multiple inputs)
        "model.onnx",  # where to save the model (can be a file or file-like object)
        export_params=True,  # store the trained parameter weights inside the model file
        opset_version=18,  # the ONNX version to export the model to
        do_constant_folding=True,  # whether to execute constant folding for optimization
        input_names=['input'],  # the model's input names
        output_names=['boxes', 'labels', 'scores'],  # the model's output names
        dynamic_axes={
            'input': {0: 'batch_size', 2: 'ims_x', 3: 'ims_y'},
            'boxes': {0: 'num_boxes'},
            'labels': {0: 'num_labels'},
            'scores': {0: 'num_scores'}
        }
    )

**Comment by shifujain** [comment]
> Facing a similar issue where an additional reshape node is getting added ![image](https://user-images.githubusercontent.com/38124154/172655468-d975b9ad-0a67-4c20-9ae0-bd2c6eb4ad30.png)
> 
> here is the problematic node ![image](https://user-images.githubusercontent.com/38124154/172655755-f4085285-688e-45f6-87c6-df91c0998c4b.png)

In my case, the issue was with the onnx conversion. It might sound funny, but **I tried changing the dummy input's batch size to any value other than 1** (batch_size=3) and fixed the max sequence and target lengths required in my use case and yes it worked!
I was then able to pass dynamic batch-sized inputs to the inference code and no reshape node errors were encountered.
